src/test/mod.rs

```diff
@@ -65,9 +65,21 @@ fn is_subpath<P>(path: &Path, subpath: &P) -> bool
 where
     P: AsRef<Path>,
 {
-    (0..path.components().count())
-        .map(|i| path.components().take(i))
-        .any(|c| c.zip(subpath.as_ref().components()).all(|(a, b)| a == b))
+    fn is_match<'a, I, J, F, T>(mut paths: I, subpath_components: F) -> bool
+    where
+        I: Iterator<Item = J>,
+        J: Iterator<Item = std::path::Component<'a>>,
+        F: Fn() -> T,
+        T: Iterator<Item = std::path::Component<'a>>,
+    {
+        paths.any(|c| c.zip(subpath_components()).all(|(a, b)| a == b))
+    }
+
+    let range = 1..=path.components().count();
+    let prefixes = range.clone().map(|i| path.components().take(i));
```

Comment:
If the first prefix doesn't match (`[foo]/bar != [bar]/baz`), then incorporating more components is useless (`[foo/bar] != [bar/baz]`), I guess?

Reply:
Yeah, you are right. I think what @calebcartwright did in https://github.com/rust-lang/rustfmt/issues/3627#issuecomment-501970476 is better than what I did first, so I have updated this PR with @calebcartwright's snippet.